src/types.ts

```diff
@@ -144,7 +144,7 @@ export type MutatorCallback<Data = any> = (
 
 export type MutatorOptions<Data = any> = {
   revalidate?: boolean
-  populateCache?: boolean | ((result: any, currentData: Data) => Data)
+  populateCache?: boolean | ((result: any, currentData?: Data) => Data)
```

Comment:
This was a suggested fix for this issue in `src/utils/mutate.ts`: https://github.com/vercel/swr/pull/1961#discussion_r872266013